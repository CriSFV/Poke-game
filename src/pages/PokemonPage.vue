<template>
  <div>
    <h1 v-if="!pokemon">Espere por favor...</h1>
    <div v-else>
      <h1>Â¿Quien es este?</h1>
      <PokemonImage :pokemonId="pokemon.id" :showPokemon="showPokemon" />
      <PokemonOptions
        :pokemons="pokemonArr"
        @poke-selection="checkAnswer($event)"
      />
      <template v-if="showAnswer">
        <h2 class="fade-in">{{ message }}</h2>
        <button @click="newGame">Nuevo Juego</button>
      </template>
    </div>
  </div>
</template>

<script>
import PokemonImage from "@/components/PokemonImage.vue";
import PokemonOptions from "@/components/PokemonOptions.vue";
import getPokemonsOptions from "@/helpers/getPokemonOptions.js";

//console.log("getPokemonsOtions, es una promesa--->", getPokemonsOptions());

export default {
  components: {
    PokemonImage,
    PokemonOptions,
  },

  data() {
    // es reactivo, Vue esta pendiente de cualquier cambio
    return {
      pokemonArr: [], // esto es loque vamos a mandar a pokemonOptions
      pokemon: null,
      showPokemon: false,
      showAnswer: false,
      message: "",
    };
  },
  methods: {
    async mixPokemonArr() {
      this.pokemonArr = await getPokemonsOptions();
      const rndInt = Math.floor(Math.random() * 4); // crea numero aleatorio para elegir un pokemon aleatorio
      this.pokemon = this.pokemonArr[rndInt];
      console.log(this.pokemon);
    },

    checkAnswer(pokemonId) {
      console.log("pokemon page", pokemonId);
      this.showAnswer = true;
      if (pokemonId === this.pokemon.id) {
        this.showPokemon = true;
        this.message = `Correcto!!`;
      } else {
        this.message = `Oppss!! el correcto es ${this.pokemon.name}`;
      }
    },
    newGame() {
      (this.pokemonArr = []), // esto es loque vamos a mandar a pokemonOptions
        (this.pokemon = null),
        (this.showPokemon = false),
        (this.showAnswer = false),
        (this.message = "");
      this.mixPokemonArr();
    },
  },
  mounted() {
    this.mixPokemonArr();
  },
};
</script>
