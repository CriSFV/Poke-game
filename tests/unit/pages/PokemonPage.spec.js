import PokemonPage from '@/pages/PokemonPage'
import { shallowMount, mount } from '@vue/test-utils'
import {pokemons} from '../mocks/pokemons.mock'


describe('PokemonPage Page', () =>{

    let wrapper
    //let mixPokemonArraySpy

    beforeEach(()=>{
       //  mixPokemonArraySpy = jest.spyOn( PokemonPage.methods, 'mixPokemonArr' )
        wrapper = shallowMount( PokemonPage )
    })

    test('debe hacer match con el snapshot',()=>{

        //expect(wrapper.html()).toMatchSnapshot()

    })

    test ('debe de llamar mixPokemonArr al montar el componente', ()=>{
        // podemos hacer el spy aquÃ­ o fuera de los test (porque se tiene que cargar antes de que se monte el wrapper)
        
        const mixPokemonArraySpy = jest.spyOn( PokemonPage.methods, 'mixPokemonArr' )
        wrapper = shallowMount( PokemonPage )
        expect( mixPokemonArraySpy ).toHaveBeenCalled()
    })

    test('debe de hacer match con el snapshot cuando carguen los pokemons, despues del mounted', ()=>{
         wrapper = shallowMount( PokemonPage, { 
            data(){
                return {
                    pokemonArr: pokemons, 
                    pokemon: null,
                    showPokemon: false,
                    showAnswer: false,
                    message: "",
                  };
            }
        } )
        expect(wrapper.html()).toMatchSnapshot()
    })

    // test('debe mostar los componentes de pokemonPicture y pokemonOptions',()=>{

    //      wrapper = shallowMount( PokemonPage, { 
    //         data(){ // esto es loque vamos a mandar a pokemonOptions
    //             return {
    //                 pokemonArr: pokemons, 
    //                 pokemon: null,
    //                 showPokemon: false,
    //                 showAnswer: false,
    //                 message: "",
    //               };
    //         }
    //     } )
    //         expect(wrapper.find('pokemon-picture-stub').exists()).toBeTruthy()
    //         //Pokemon picture de de existir
    //         //Pokemon picture pokeonId sea igual a 1 (posicion 0)
    //         //Pokemon option de de existir. componente tobe(true)
    //         //Pokemon optition tenga atributo pokemons toBe(true)

    // })

})