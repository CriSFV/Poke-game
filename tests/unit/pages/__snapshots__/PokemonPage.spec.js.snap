// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Page debe de hacer match con el snapshot cuando carguen los pokemons, despues del mounted 1`] = `
<div>
  <h1>Espere por favor...</h1>
</div>
`;

exports[`PokemonPage Page debe hacer match con el snapshot 1`] = `
<div>
  <h1>Espere por favor...</h1>
</div>
`;
